### PR TITLE
Add ability to skip SELinux python bindings checks (#34340).

### PR DESCRIPTION
**Pull request by estheruary**
##### SUMMARY

Adds the ability to run Ansible modules on hosts without the SELinux python bindings. For me, this fixes the issue of having to drop to raw modules in order to prep a system before that package can be installed.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

Adds the ability for users to specify the `seignore` option on modules that perform file operations that are SELinux sensitive.

##### ADDITIONAL INFORMATION

Relevant to issue #34340.

### CODE REVIEW

File: lib/ansible/modules/monitoring/sensu_check.py [hunk 1]
```diff
@@ -132,6 +132,12 @@
     version_added: "2.1"
     description:
       - The check source, used to create a JIT Sensu client for an external resource (e.g. a network switch).
+  seignore:
+    description:
+      - Ignore SELinux for purposes of manipulating file contexts.
+    type: bool
+    required: false
```
Comment (dagwieers) [inline lib/ansible/modules/monitoring/sensu_check.py]:
```suggestion
```

File: lib/ansible/modules/monitoring/sensu_check.py [hunk 2]
```diff
@@ -132,6 +132,12 @@
...
+    default: 'no'
```
Comment (dagwieers) [inline lib/ansible/modules/monitoring/sensu_check.py]:
```suggestion
    default: no
```

File: lib/ansible/modules/monitoring/sensu_check.py [hunk 3]
```diff
@@ -338,6 +344,7 @@ def main():
                 'high_flap_threshold': {'type': 'int'},
                 'custom': {'type': 'dict'},
                 'source': {'type': 'str'},
+                'seignore': {'type': 'bool', 'default': 'no'},
```
Comment (dagwieers) [inline lib/ansible/modules/monitoring/sensu_check.py]:
```suggestion
                'seignore': {'type': 'bool', 'default': False},
```

File: lib/ansible/modules/monitoring/sensu_subscription.py [hunk 4]
```diff
@@ -136,6 +142,7 @@ def main():
                 'path': {'type': 'str', 'default': '/etc/sensu/conf.d/subscriptions.json'},
                 'state': {'type': 'str', 'default': 'present', 'choices': ['present', 'absent']},
                 'backup': {'type': 'bool', 'default': 'no'},
+                'seignore': {'type': 'bool', 'default': 'no'},
```
Comment (dagwieers) [inline lib/ansible/modules/monitoring/sensu_subscription.py]:
```suggestion
                'seignore': {'type': 'bool', 'default': False},
```

File: lib/ansible/modules/network/f5/bigip_ucs_fetch.py [hunk 5]
```diff
@@ -142,6 +142,11 @@
   returned: success
   type: int
   sample: 1220
+seignore:
+  description: Ignore SELinux for purposes of manipulating file contexts
+  type: bool
+  required: false
+  default: false
```
Comment (dagwieers) [inline lib/ansible/modules/network/f5/bigip_ucs_fetch.py]:
```suggestion
  default: no
```

File: lib/ansible/modules/network/f5/bigip_ucs_fetch.py [hunk 6]
```diff
@@ -142,6 +142,11 @@
   returned: success
   type: int
   sample: 1220
+seignore:
+  description: Ignore SELinux for purposes of manipulating file contexts
+  type: bool
+  required: false
```
Comment (dagwieers) [inline lib/ansible/modules/network/f5/bigip_ucs_fetch.py]:
```suggestion
```

File: lib/ansible/modules/network/f5/bigip_ucs_fetch.py [hunk 7]
```diff
@@ -142,6 +142,11 @@
   returned: success
   type: int
   sample: 1220
+seignore:
+  description: Ignore SELinux for purposes of manipulating file contexts
```
Comment (dagwieers) [inline lib/ansible/modules/network/f5/bigip_ucs_fetch.py]:
```suggestion
  description: Ignore SELinux for purposes of manipulating file contexts.
```

File: lib/ansible/modules/packaging/os/apt_repository.py [hunk 8]
```diff
@@ -61,6 +61,12 @@
             - Override the distribution codename to use for PPA repositories.
               Should usually only be set when working with a PPA on a non-Ubuntu target (e.g. Debian or Mint)
         version_added: '2.3'
+    seignore:
+        description:
+            - Ignore SELinux for purposes of manipulating file contexts.
+        type: bool
+        required: false
```
Comment (dagwieers) [inline lib/ansible/modules/packaging/os/apt_repository.py]:
```suggestion
```

File: lib/ansible/modules/packaging/os/apt_repository.py [hunk 9]
```diff
@@ -61,6 +61,12 @@
...
+        default: false
```
Comment (dagwieers) [inline lib/ansible/modules/packaging/os/apt_repository.py]:
```suggestion
        default: no
```

File: lib/ansible/modules/crypto/openssl_dhparam.py [hunk 10]
```diff
@@ -46,6 +46,12 @@
             - Name of the file in which the generated parameters will be saved.
         type: path
         required: true
+    seignore:
+        description:
+            - Ignore SELinux for purposes of manipulating file contexts.
+        type: bool
+        required: false
```
Comment (dagwieers) [inline lib/ansible/modules/crypto/openssl_dhparam.py]: Please do this everywhere...
```suggestion
```

File: lib/ansible/modules/crypto/openssl_dhparam.py [hunk 11]
```diff
@@ -46,6 +46,12 @@
...
+        default: false
```
Comment (dagwieers) [inline lib/ansible/modules/crypto/openssl_dhparam.py]: Please do this everywhere.
```suggestion
        default: no
```

**Comment by wojtek0806** [comment]
@estheruary

Looking at the changes to bigip_ucs_fetch module it seems you have not updated the docs too, at least it seems so from looking at the latest commit, please address this

**Comment by samdoran** [comment]
!component =/lib/ansible/module_utils/basic.py

**Pull request closed without merging**